spec/unit/provider/package/dpkg_spec.rb

```diff
@@ -221,19 +221,28 @@ def parser_test(dpkg_output_string, gold_hash, number_of_debug_logs = 0)
       allow(Tempfile).to receive(:new).and_return(tempfile)
     end
 
-    it "installs first if holding" do
+    it "installs first if package is not present and ensure holding" do
+
       allow(provider).to receive(:execute)
+      allow(provider).to receive(:package_not_installed?).and_return(false)
       expect(provider).to receive(:install).once
       provider.hold
     end
 
+    it "skips install new package if package is allready installed" do
+      allow(provider).to receive(:package_not_installed?).and_return(true)
+      expect(provider).not_to receive(:install)
+    end
```

Comment:
Missing `provider.hold` here? As-is the reason `provider.install` isn't being called is because nothing is being called in the test.

Reply:
@jhelwig good catch!
Added provider hold and provider to recieve execute in order to test this